### PR TITLE
Graphics3d breaking changes

### PR DESCRIPTION
**Pull request by TiagoCavalcanteTrindade**
image

### CODE REVIEW

File: mathics/builtin/box/graphics3d.py [hunk 1]
```diff
@@ -384,14 +375,6 @@ def boxes_to_json(self, leaves=None, **options):
                     "ticks": ticks,
                     "ticks_style": js_ticks_style,
                 },
-                "extent": {
```
Comment (rocky) [inline mathics/builtin/box/graphics3d.py:387]: Can't you just ignore it in the other side?

File: mathics/builtin/box/graphics3d.py [hunk 2]
```diff
@@ -63,40 +63,40 @@ def _prepare_elements(self, leaves, options, max_width=None):
 
         if lighting == "System`Automatic":
             self.lighting = [
-                {"type": "Ambient", "color": [0.3, 0.2, 0.4]},
+                {"type": "ambient", "color": [0.3, 0.2, 0.4]},
```
Comment (rocky) [inline mathics/builtin/box/graphics3d.py:66]: Isn't this handled by the transformation layer in mathics-graphics3d-backend ?

File: mathics/builtin/box/graphics3d.py (hunk 1 above)
Comment (TiagoCavalcanteTrindade) [reply mathics/builtin/box/graphics3d.py:387]: Yes, but why keep it there?

File: mathics/builtin/box/graphics3d.py (hunk 2 above)
Comment (TiagoCavalcanteTrindade) [reply mathics/builtin/box/graphics3d.py:66]: Yes, but is weird to have primitive names lower-case and light names capital-case.

Comment (rocky) [reply mathics/builtin/box/graphics3d.py:66]: I wasn't saying this is wrong - just that mathics-threejs-backend _also_ needs to be able to handle things like this. 

The current plan is to release Mathics Django as is. And then _after_ that settles we will a slew of API breaking changes.

In general things, have breaking way too much recently. We need more control everywhere on limiting  breakage.

Comment (TiagoCavalcanteTrindade) [reply mathics/builtin/box/graphics3d.py:66]: Ok, so just remove the viewpoint scaling by now as it isn't a breaking change?

File: mathics/builtin/box/graphics3d.py (hunk 1 above)
Comment (rocky) [reply mathics/builtin/box/graphics3d.py:387]: Same as above. It will change and should. But not before the Mathics-Django release.

Comment (rocky) [reply mathics/builtin/box/graphics3d.py:387]: Probably a good way to approach this is to start going over the API doc in mathics-threejs-backend review that and then come back here to make it match that specification. 

But all of this is major API upheaval. I'd like to have at least one more stable release before that.

File: mathics/builtin/box/graphics3d.py (hunk 2 above)
Comment (rocky) [reply mathics/builtin/box/graphics3d.py:66]: With all of the changes it is hard to follow this PR.  It feels like after things settle down we should plain on a major Graphics3D in compact JSON design.  

Personally, I think that all of the design work and implementation in mathics-threejs-backend could go on now if you are so disposed. Nothing prevents creating whatever API  is best and converting what we have now to that.

In the PRs here in the short term we should separate breaking API from non-breaking. 

Adding fields to existing JSON is fine that isn't going to break older Mathics-Django that happen use Mathics inore dev. changing existing values would.  And having mathics-threejs-backend ignore existing fields is fine too.
